app/src/main/java/com/example/homework/parse.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.homework
+
+fun parse(str: String, calculator: Calculator): String {
+    var returnString = ""
+    var operation = ""
+    var firstNumber = ""
+    var secondNumber = ""
+    when {
+        str.contains("*") -> {
+            operation = Regex("[\\d]+\\*[\\d]+").find(str)?.groups?.get(0)?.value!!
+            firstNumber =
+                Regex("[\\d]+\\*").find(operation!!)!!.groups[0]!!.value.replace("*", "")
+            secondNumber =
+                Regex("\\*[\\d]+").find(operation!!)!!.groups[0]!!.value.replace("*", "")
+            calculator.mul(firstNumber.toInt(), secondNumber.toInt())
+            returnString = str.replace(operation, "")
+
+        }
+        str.contains("/") -> {
+            operation = Regex("[\\d]+\\/[\\d]+").find(str)?.groups?.get(0)?.value!!
```

Comment:
Явная копипаста, можно было сделать что-нибудь такое
```suggestion
            operation = Regex("[\\d]+\\/[\\d]+").findFirstMatching(str)
```